mlir/lib/Dialect/SCF/Utils/Utils.cpp

```diff
@@ -864,6 +864,18 @@ LogicalResult mlir::coalesceLoops(RewriterBase &rewriter,
 
     Operation *innerTerminator = innerLoop.getBody()->getTerminator();
     auto yieldedVals = llvm::to_vector(innerTerminator->getOperands());
+    assert(llvm::equal(outerLoop.getRegionIterArgs(), innerLoop.getInitArgs()));
+    for (Value &yieldedVal : yieldedVals) {
+      // The yielded value may be an iteration argument of the inner loop
+      // which is about to be inlined.
+      auto iter = llvm::find(innerLoop.getRegionIterArgs(), yieldedVal);
+      if (iter != innerLoop.getRegionIterArgs().end()) {
+        unsigned iterArgIndex = iter - innerLoop.getRegionIterArgs().begin();
+        // `outerLoop` iter args identical to the `innerLoop` init args.
```

Comment:
I think this is better to just return the corresponding `iter_args` operand of the inner loop?

Reply:
Thanks for the response, did you mean the `init_args` operand? because the iter_args are the one we want to replace them.

Reply:
Basically, I think instead of 
```
yieldedVal = outerLoop.getRegionIterArgs()[..]
```

it should be
```
yieldedVal = innerLoop.getInitArgs()[...]
```

Sorry, the IR format and method names are confusing...

Reply:
Great, fixed it.